packages/plugins/output/src/index.ts

```diff
@@ -0,0 +1,146 @@
+                children: true,
+                chunks: true,
+                chunkGroupAuxiliary: true,
+                chunkGroupChildren: true,
+                chunkGroups: true,
+                chunkModules: true,
+                chunkRelations: true,
+                entrypoints: true,
+                errors: true,
+                ids: true,
+                modules: true,
+                nestedModules: true,
+                reasons: true,
+                relatedAssets: true,
+                warnings: true,
+            });
+
+            write(statsJson);
+        });
+    };
+
+const getRollupPlugin = (
+    write: (outputs: OutputBundle[]) => void,
+): PluginOptions['rollup'] & PluginOptions['vite'] => {
+    const outputs: Set<OutputBundle> = new Set();
+    return {
+        buildStart() {
+            // Clear set on build start.
+            outputs.clear();
+        },
+        writeBundle(opts, bundle) {
+            outputs.add(bundle);
+        },
+        closeBundle() {
+            write(Array.from(outputs));
+        },
+    };
+};
+
+export const getFilePath = (outDir: string, pathOption: string, filename: string): string => {
+    // If we have an absolute path, we use it as is.
+    const outputPath = path.isAbsolute(pathOption)
+        ? pathOption
+        : // Otherwise, we resolve it relative to the bundler output directory.
+          path.resolve(outDir, pathOption);
+    return path.resolve(outputPath, filename);
+};
+
+export const getPlugins: GetPlugins = ({ options, context }) => {
+    // Verify configuration.
+    const validatedOptions = validateOptions(options);
+
+    // If the plugin is not enabled, return an empty array.
+    if (!validatedOptions.enable) {
+        return [];
+    }
+
+    const writeFile = (name: FileKey, data: any) => {
+        const fileValue: FileValue = validatedOptions.files[name];
+        if (data && fileValue !== false) {
+            outputJsonSync(
+                getFilePath(context.bundler.outDir, validatedOptions.path, fileValue),
+                data,
+            );
+        }
+    };
+
+    return [
+        {
+            name: PLUGIN_NAME,
+            buildReport(report) {
+                const serializedReport = serializeBuildReport(report);
+                writeFile('build', {
+                    bundler: serializedReport.bundler,
+                    metadata: serializedReport.metadata,
+                    start: serializedReport.start,
+                    end: serializedReport.end,
+                    duration: serializedReport.duration,
+                    writeDuration: serializedReport.writeDuration,
+                    entries: serializedReport.entries,
+                    outputs: serializedReport.outputs,
+                });
+                writeFile('logs', serializedReport.logs);
+                writeFile('timings', serializedReport.timings);
+                writeFile('dependencies', serializedReport.inputs);
```

Comment:
just to make sure, no typo here?

Reply:
No typo, all good, it just happens that the `inputs` list of the build report also acts as the dependency graph of the project.